### DISCUSSION TITLE
python numpy (isin and unique)

### DISCUSSION THREAD

**Discussion by pengzeli** [libvips/libvips#4218] [Q&A]
Is there an operation similar to numpy -- np.isin and np.unique ?
I want to perform this operation，
`remove_id = np.unique(mask1)`
`mask2 = ~np.isin(mask2, remove_id) * mask2`

**Comment by jcupitt** [comment]
Hello @pengzeli,

No, but you can use numpy on libvips images, perhaps:

```python
mask1 = some pyvips image
remove_id = np.unique(mask1.numpy())
mask2 = ~np.isin(mask2, remove_cell_id) * mask2
mask3 = pyvips.Image.new_from_array(mask2)
```

The array memory will be shared efficiently between numpy and pyvips.

**Comment by pengzeli** [comment]
I got it, thanks for the response!
